python/cudf/tests/test_orc.py

```diff
@@ -77,6 +108,26 @@ def test_orc_reader_filenotfound(tmpdir):
         cudf.read_orc(tmpdir.mkdir("cudf_orc"))
 
 
+def test_orc_reader_local_filepath():
+    path = '~/TestOrcFile.test1.orc'
```

Comment:
This doesn't look right, should this be datadir similar to other tests?

Reply:
PANDAS' allows giving a filepath of `~/xxx.csv` and it'll auto-expand to `/home/jieong/xxx.csv`. 

This PR adds that user home path expansion functionality but not sure how to test in CI. I would need to write a temporary file to the CI machine's user home directory and read it to actually exercise it...is that possible? Right now, this test is skipped in CI. Or any other idea?

Reply:
No, `datadir` will point to `python/cudf/tests/data` which is where I would expect these files to live. Is that not correct?

Reply:
Hrmmm, so `~` doesn't contain part of `datadir` on the CI machines?

PANDAS expands a file path of `~/xxx.csv` to `/home/jieong/xxx.csv`. This PR adds that functionality too, so I want a test case for it...

Reply:
Is this specifically for testing that the `~` character is handled in filepaths passed down to the C++ layer?

Reply:
The expansion is done in the Python layer currently, in `get_filepath_or_buffer()` which is similar to where PANDAS does it.

Reply:
Alternative workarounds here aren't very good so I guess lets just leave this as is. Do we want to change the filename to something more generic for both here and the Parquet file? Below it's named `spark_timestamp.snappy.parquet`

Reply:
Done!